### PR TITLE
Fix for timeouts when tests are run on system under high load

### PR DESCRIPTION
**Pull request by hirenvadalia**
#### Describe Bug or Feature
* When the system is under high load, PBS stop/restart operation takes time to gracefully shutdown but PBS init script only waits for 30 sec for graceful shutdown of PBS daemons and exit with error after 30 sec if PBS daemons are not shutdown
* PTL has the capability to override default max attempts and the interval between attempts but as of now, that overridden value is only used Server.expect method, but in PTL there are iterator methods like PBSService._log_match, *.isUp, etc... and use overridden value in these methods will help when tests are run on the system under high load
* User can override default test case timeout via pbs_benchpress but PTL doesn't consider that if a test case has timeout applied via `@timeout` decorator and overridden value is greater than test case timeout
* if log_match is called with existence=False then even if the log message is not found still log_match iterates till max attempts reached

#### Describe Your Change
* Increased wait time to 180 sec from 30 sec in pbs_init.d, i.e. wait for 180 sec for PBS daemons to graceful shutdown
* Made max attempts and attempt interval generic for all PBS objects and use it all iterator method
* Modified get timeout method in runner plugins such a way that it will return max(default_testcase_timeout, testcase_timeout)
* Fixed log_match not iterate again if the message is not found when existence is False

### CODE REVIEW

**Comment by hirenvadalia** [comment]
@PBSPro/pbspro-maintainers please do needful for this PR... thanks

File: test/fw/ptl/lib/pbs_testlib.py [hunk 1]
```diff
@@ -3354,9 +3390,17 @@ class PBSObject(object):
...
     """
 
+    logger = logging.getLogger(__name__)
     utils = BatchUtils()
     platform = sys.platform
 
+    ptl_conf = {
+        'mode': PTL_API,
+        'max_attempts': 60,
+        'attempt_interval': 0.5,
```
Comment (mike0042) [inline test/fw/ptl/lib/pbs_testlib.py]: Should these values be set based on object.options['PTL_MAX_ATTEMPTS'] and object.options['PTL_ATTEMPT_INTERVAL'] as opposed to hardcoding them? I noticed that the same values are defined in two places.

Comment (hirenvadalia) [reply test/fw/ptl/lib/pbs_testlib.py]: These are default values if a user or any PTL command doesn't call `PtlConfig()` and options dict is only accessible in PtlConfig class not in PBSObject or its child

Comment (hirenvadalia) [reply test/fw/ptl/lib/pbs_testlib.py]: @mike0042 Sorry for pushing but this PR is little urgent for other dependent work so if you don't have any further comment then can you please do needfull...

**Pull request merged**